cirq/api/google/v2/program.proto

```diff
@@ -93,14 +93,12 @@ message Operation {
   // Which gate this operation corresponds to.
   Gate gate = 1;
 
-  // Arguments needed to fully specify the gate.
-  repeated Arg args = 2;
+  // Map from the argument name to the Argument needed to fully specify
+  // the gate.
+  map<string, Arg> args = 2;
 
   // Which qubits the operation acts on.
   repeated Qubit qubits = 3;
-
-  // If the operation produces results, these are referenced by these keys.
-  repeated string keys = 4;
```

Comment:
So I do think putting it as an arg makes things much cleaner, and it fits into how this is used while deserializing quite nicely.  

But what you are describing would be nice., to be able to determine the amount of information this gate will provide.  My first thought would be to record this separately.  One worry I have is that will for computational basis state measurements this is a simple "this gate produces this much data", if we had an "expectation value" type gate this will then need to know repetitions / parameter counts (which is fine, I guess).  But more worrisome is that we might have an argument to an expectation value gate that changes the amount of info we return back.